demos/mqtt/mqtt_demo_connection_sharing/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1532 @@
+    }
+    else
+    {
+        /* The client is now connected to the broker. Subscribe to the passed topic
+         * by sending a subscribe packet. This client will then publish to the topic(s)
+         * related to the subscribed topic filter, so it will expect all the messages it sends to
+         * the broker to be sent back to it from the broker. The subscription manager would
+         * then dispatch all messages from the incoming PUBLISH topic to the above registered
+         * callback. This demo uses QOS1 in Subscribe, therefore, the Publish messages received
+         * from the broker will have QOS1. */
+        LogInfo( ( "Subscribing to the MQTT topic %.*s.",
+                   topicFilterLength,
+                   pTopicFilter ) );
+
+        returnStatus = subscribeToTopic( pContext,
+                                         pTopicFilter,
+                                         topicFilterLength );
+    }
+
+    if( returnStatus != EXIT_SUCCESS )
+    {
+        /* Remove the registered callback for the temperature topic filter as
+        * the subscription operation for the topic filter did not succeed. */
+        ( void ) SubscriptionManager_RemoveCallback( pTopicFilter,
+                                                     topicFilterLength );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int unsubscribeFromTopic( MQTTContext_t * pMqttContext,
+                                 const char * pTopicFilter,
+                                 uint16_t topicFilterLength )
+{
+    int returnStatus = EXIT_SUCCESS;
+    MQTTStatus_t mqttStatus;
+    MQTTSubscribeInfo_t pSubscriptionList[ 1 ];
+
+    assert( pMqttContext != NULL );
+
+    /* Start with everything at 0. */
+    ( void ) memset( ( void * ) pSubscriptionList, 0x00, sizeof( pSubscriptionList ) );
+
+    pSubscriptionList[ 0 ].pTopicFilter = pTopicFilter;
+    pSubscriptionList[ 0 ].topicFilterLength = topicFilterLength;
+
+    /* Generate packet identifier for the UNSUBSCRIBE packet. */
+    globalUnsubscribePacketIdentifier = MQTT_GetPacketId( pMqttContext );
+
+    /* Send UNSUBSCRIBE packet. */
+    mqttStatus = MQTT_Unsubscribe( pMqttContext,
+                                   pSubscriptionList,
+                                   sizeof( pSubscriptionList ) / sizeof( MQTTSubscribeInfo_t ),
+                                   globalUnsubscribePacketIdentifier );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        LogError( ( "Failed to send UNSUBSCRIBE packet to broker with error = %u.",
+                    mqttStatus ) );
+        returnStatus = EXIT_FAILURE;
+    }
+    else
+    {
+        LogInfo( ( "UNSUBSCRIBE sent for topic %.*s to broker.\n\n",
+                   topicFilterLength,
+                   pTopicFilter ) );
+
+        /* Process Incoming UNSUBACK packet from the broker. */
+        mqttStatus = MQTT_ProcessLoop( pMqttContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+
```

Comment:
Would it be useful to have a global `receivedUnsuback` flag? If we do, we don't have to wait the entire timeout and can just wait until we receive the unsuback. If not, I don't really see why this function needs to call the process loop. Since we have three consecutive `unsubscribeFromTopic` calls, It would save 2 * MQTT_PROCESS_LOOP_TIMEOUT_MS milliseconds If we only called the process loop once after all three calls.

Reply:
I can instead unsubscribe from all topic filters in the same UNSUBSCRIBE request

Reply:
Have updated the private function to take in a list of topic filters to unsubscribe with a single packet.